bob-apps/bin/bobd.rs

```diff
@@ -25,35 +25,65 @@ use std::fs::create_dir;
 extern crate log;
 
+use log4rs::append::console::ConsoleAppender;
+use log4rs::encode::pattern::PatternEncoder;
+use log4rs::config::{Appender, Config, Root};
+
+use network_interface::{NetworkInterface, NetworkInterfaceConfig, Addr};
+
+use anyhow::{anyhow, Context, Result as AnyResult};
+
 #[tokio::main]
 async fn main() {
     let matches = get_matches();
 
-    if matches.value_of("cluster").is_none() {
-        eprintln!("Expect cluster config");
-        eprintln!("use --help");
-        return;
-    }
+    let cluster;
+    let node;
+    if let (sc, Some(sub_matches)) = matches.subcommand() {
+        match sc {
+            "testmode" => match configure_testmode(sub_matches) {
+                Ok((c, n)) => {
+                    cluster = c;
+                    node = n;
+                }
+                Err(e) => {
+                    eprintln!("Initialization error: {}", e);
+                    eprintln!("use --help");
+                    return;
+                }
+            },
+            _ => unreachable!("unknown command"),
+        }
+    } else {
+        if matches.value_of("cluster").is_none() {
+            eprintln!("Expect cluster config");
+            eprintln!("use --help");
+            return;
+        }
 
-    if matches.value_of("node").is_none() {
-        eprintln!("Expect node config");
-        eprintln!("use --help");
-        return;
-    }
+        if matches.value_of("node").is_none() {
+            eprintln!("Expect node config");
+            eprintln!("use --help");
+            return;
+        }
+
+        let cluster_config = matches.value_of("cluster").unwrap();
+        println!("Cluster config: {:?}", cluster_config);
+        cluster = ClusterConfig::try_get(cluster_config).await.map_err(|err| {
+            eprintln!("Cluster config parsing error: {}", err);
+            err
+        }).expect("Cluster config parsing error");
 
-    let cluster_config = matches.value_of("cluster").expect("'cluster' argument is required");
-    println!("Cluster config: {:?}", cluster_config);
-    let cluster = ClusterConfig::try_get(cluster_config).await.map_err(|err| {
-        eprintln!("Cluster config parsing error: {}", err);
-        err
-    }).expect("Cluster config parsing error");
 
-    let node_config_file = matches.value_of("node").expect("'node' argument is required");
-    println!("Node config: {:?}", node_config_file);
-    let node = cluster.get(node_config_file).await.map_err(|err| {
-        eprintln!("Node config parsing error: {}", err);
-        err
-    }).expect("Node config parsing error");
+        let node_config_file = matches.value_of("node").unwrap();
+        println!("Node config: {:?}", node_config_file);
+        node = cluster.get(node_config_file).await.map_err(|err| {
+            eprintln!("Node config parsing error: {}", err);
+            err
+        }).expect("Node config parsing error");
+
+        check_folders(&node, matches.is_present("init_folders"));
```

Comment:
`check_folders` is also called below. Looks like the call below should be removed, because `init_folders` flag is only valid here